src/main/java/de/tum/in/www1/exerciseapp/domain/util/FileManagement.java

```diff
@@ -96,21 +96,24 @@ private static String actualPathForPublicPath(String filePath) {
      * @param entityId   the id of the entity associated with the file (may be null)
      * @return the public path that can be used by users to access the file from outside
      */
-    private static String publicPathForActualPath(String actualPath, Long entityId) {
+    private static String publicPathForActualPath(Path actualPath, Long entityId) {
         // first extract filename
-        String filename = actualPath.substring(actualPath.lastIndexOf("/") + 1);
+        String filename = actualPath.getFileName().toString();
 
         // generate part for id
         String id = entityId == null ? Constants.FILEPATH_ID_PLACHEOLDER : entityId.toString();
 
+        // convert path to unix style path (to compare with Constants)
+        String pathInUnix = FilenameUtils.separatorsToUnix(actualPath.toString());
```

Comment:
Is this really a good idea? why do you not use OS specific separators in all places using "File.separator"?